File: core/src/main/java/forpleuvoir/dhwu1a/core/message/messageitem/Dice.java
package forpleuvoir.dhwu1a.core.message.messageitem;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem;
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType;

import static forpleuvoir.dhwu1a.core.common.ApiKey.VALUE;

/**
 * 骰子消息
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.messageitem
 * <p>#class_name Dice
 * <p>#create_time 2021/7/1 1:08
 */
public class Dice extends MessageItem {
    /**
     * 点数
     */
    @SerializedName(VALUE)
    public Integer value;

    public Dice(Integer value) {
        super(MessageItemType.Dice);
        this.value = value;
    }

    @Override
    public String toPlainText() {
        return String.format("骰子:%d点", value);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/messageitem/Dice.kt
package forpleuvoir.dhwu1a.core.message.messageitem

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.*
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItem
import forpleuvoir.dhwu1a.core.message.messageitem.base.MessageItemType

/**
 * 骰子消息
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.messageitem
 *
 * #class_name Dice
 *
 * #create_time 2021/7/1 1:08
 */
class Dice(
    /**
     * 点数
     */
    @field:SerializedName(VALUE) val value: Int
) : MessageItem(MessageItemType.Dice) {
    override fun toPlainText(): String {
        return String.format("骰子:%d点", value)
    }
}